community/common/src/main/java/org/neo4j/kernel/Health.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel;
+
+public interface Health
+{
+    /**
+     * Asserts that the database is in good health. If that is not the case then the cause of the
+     * unhealthy state is wrapped in an exception of the given type, i.e. the panic disguise.
+     *
+     * @param panicDisguise the cause of the unhealthy state wrapped in an exception of this type.
+     * @throws EXCEPTION exception type to wrap cause in.
+     */
+    default <EXCEPTION extends Throwable> void assertHealthy( Class<EXCEPTION> panicDisguise ) throws EXCEPTION
```

Comment:
To simplify `Adapter` implementation. Will remove `Adapter` and create more descriptive class as suggested 👍